cmd/skaffold/app/cmd/run_test.go

```diff
@@ -62,6 +62,10 @@ func (r *mockRunRunner) Build(_ context.Context, _ io.Writer, artifacts []*lates
 	return result, nil
 }
 
+func (r *mockRunRunner) Test(context.Context, io.Writer, []build.Artifact) error {
```

Comment:
I think you could have a simpler test modelled on `TestBuildImageFlag`: modify the `mockRunRunner` to record whether `Test()` and `DeployAndLog()` methods were run, and what should be the result of the `Test()` method.

Reply:
Updated unit test to validate Test() and DeployAndLog() methods.